mantidimaging/gui/windows/load_dialog/presenter.py

```diff
@@ -123,7 +123,11 @@ def do_update_flat_or_dark(self, field: Field, name: str, suffix: str):
         if not selected_file:
             return
         selected_dir = Path(os.path.dirname(selected_file))
-        field.set_images(find_images(selected_dir, name, suffix, image_format=self.image_format, logger=logger))
+        images = find_images(selected_dir, name, suffix, image_format=self.image_format, logger=logger)
+        if not images:
+            base_name = os.path.basename(selected_file).split("_")[0]
```

Comment:
Can you change this to split on the last `_`
`os.path.basename(selected_file).rpartition("_")[0]`

This means that if the filename is `spheres_bbb_0000.tiff` then it won't match `spheres_ccc_0000.tiff`

(I added a strange_names2 file, currently when I select `spheres_bbb_0000.tiff` for Flat Before, it then matches all 409 images)